pkg/cache/clusterqueue_snapshot.go

```diff
@@ -62,43 +67,29 @@ func (c *ClusterQueueSnapshot) Fits(frq resources.FlavorResourceQuantities) bool
 }
 
 func (c *ClusterQueueSnapshot) QuotaFor(fr resources.FlavorResource) *ResourceQuota {
-	return c.Quotas[fr]
+	return c.ResourceNode.Quotas[fr]
 }
 
 func (c *ClusterQueueSnapshot) Borrowing(fr resources.FlavorResource) bool {
 	return c.BorrowingWith(fr, 0)
 }
 
 func (c *ClusterQueueSnapshot) BorrowingWith(fr resources.FlavorResource, val int64) bool {
-	return c.Usage[fr]+val > c.nominal(fr)
+	return c.usageFor(fr)+val > c.QuotaFor(fr).nominalSafe()
 }
 
+// Available returns the current capacity available, before preempting
+// any workloads. Includes local capacity and capacity borrowed from
+// Cohort.
 func (c *ClusterQueueSnapshot) Available(fr resources.FlavorResource) int64 {
-	if c.Cohort == nil {
-		return max(0, c.nominal(fr)-c.Usage[fr])
-	}
-	capacityAvailable := c.RequestableCohortQuota(fr) - c.UsedCohortQuota(fr)
-
-	// if the borrowing limit exists, we cap our available capacity by the borrowing limit.
-	if borrowingLimit := c.borrowingLimit(fr); borrowingLimit != nil {
-		withBorrowingRemaining := c.nominal(fr) + *borrowingLimit - c.Usage[fr]
-		capacityAvailable = min(capacityAvailable, withBorrowingRemaining)
-	}
-	return max(0, capacityAvailable)
-}
-
-func (c *ClusterQueueSnapshot) nominal(fr resources.FlavorResource) int64 {
-	if quota := c.QuotaFor(fr); quota != nil {
-		return quota.Nominal
-	}
-	return 0
+	return available(c, fr, true)
 }
 
-func (c *ClusterQueueSnapshot) borrowingLimit(fr resources.FlavorResource) *int64 {
-	if quota := c.QuotaFor(fr); quota != nil {
-		return quota.BorrowingLimit
-	}
-	return nil
+// MaxAvailable returns the largest workload this ClusterQueue could
+// possibly admit, accounting for its capacity and capacity borrowed
+// its from Cohort.
+func (c *ClusterQueueSnapshot) MaxAvailable(fr resources.FlavorResource) int64 {
```

Comment:
```suggestion
func (c *ClusterQueueSnapshot) PotentialAvailable(fr resources.FlavorResource) int64 {
```
or `AvailableIfEmpty`?

When reading the changes in `flavorassigner.go`, it's not obvious what the difference is between `Available` and `MaxAvailable`.